Libraries/Opc.Ua.Client/NodeCache.cs

```diff
@@ -117,24 +139,33 @@ public IList<INode> Find(IList<ExpandedNodeId> nodeIds)
             IList<INode> nodes = new List<INode>(count);
             var fetchNodeIds = new ExpandedNodeIdCollection();
 
-            int ii;
-            for (ii = 0; ii < count; ii++)
+            try
             {
-                // check if node already exists.
-                INode node = m_nodes.Find(nodeIds[ii]);
+                int ii;
 
-                // do not return temporary nodes created after a Browse().
-                if (node != null &&
-                    node?.GetType() != typeof(Node))
-                {
-                    nodes.Add(node);
-                }
-                else
+                m_cacheLock.EnterReadLock();
```

Comment:
Could this be more fine grained lock? per iteration?

Reply:
changed it